packages/backend/src/services/UserAttributesService/UserAttributeUtils.ts

```diff
@@ -1,20 +1,18 @@
-import {
-    CompiledTable,
-    Dimension,
-    Explore,
-    UserAttribute,
-} from '@lightdash/common';
+import { Explore, UserAttribute } from '@lightdash/common';
 
 export const hasUserAttribute = (
     userAttributes: UserAttribute[],
     attributeName: string,
     value: string,
 ) =>
-    userAttributes.some(
-        (ua) =>
-            ua.name === attributeName &&
-            ua.users.some((u) => u.value === value),
-    );
+    userAttributes.some((ua) => {
```

Comment:
Is this for column level security? So we're checking if any user attribute has a matching key *and* value to the provided `attributeName` and `value` ?

Is it meant to be valid for any user or the specific user? This looks like it returns true if any user has been assigned  the argument`value`. This also duplicates the logic a bit of ```userValue || defaultValue```

Reply:
This is used on column level security, on query build and global search

I added a new collection of unit tests on `UserAttributeUtil` about how it works :

I think this logic is the most interesting one: 

> should be false if attribute value does not match even if default is present